src/Compilers/VisualBasic/Portable/Binding/Binder_Statements.vb

```diff
@@ -412,6 +412,37 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 statements.Add(New BoundReturnStatement(methodBlock.EndBlockStatement, Nothing, Nothing, Nothing))
             End If
 
+            ' Inject implicit base constructor call, if appropriate
+            If Not methodSymbol.IsImplicitlyDeclared AndAlso methodSymbol.MethodKind = MethodKind.Constructor Then
+                Dim referencedConstructor As MethodSymbol = Nothing
+                Dim injectDefaultConstructorCall As Boolean = False
+
+                MethodCompiler.GetExplicitlyOrImplicitlyReferencedConstructor(methodSymbol, statements(0), Me,
+                                                                              diagnostics, referencedConstructor, injectDefaultConstructorCall)
+
+                ' If we didn't find explicitly referenced constructor, use implicitly generated call
+                If injectDefaultConstructorCall Then
+                    If referencedConstructor IsNot Nothing Then
+                        Dim initializer As BoundExpressionStatement = MethodCompiler.BindDefaultConstructorInitializer(methodSymbol, referencedConstructor, diagnostics, Me)
+                        Debug.Assert(initializer.Expression.Kind = BoundKind.Call OrElse
+                                     (initializer.HasErrors AndAlso
+                                      (Not diagnostics.AccumulatesDiagnostics OrElse diagnostics.HasAnyResolvedErrors)))
+                        statements.Insert(0, initializer)
+                    Else
+                        Debug.Assert(Not diagnostics.AccumulatesDiagnostics OrElse diagnostics.HasAnyResolvedErrors OrElse If(methodSymbol.ContainingType.BaseTypeNoUseSiteDiagnostics?.IsErrorType(), False))
```

Comment:
>If(methodSymbol.ContainingType.BaseTypeNoUseSiteDiagnostics?.IsErrorType(), False) [](http://example.com/codeflow?start=123&length=82)

Does this mean `methodSymbol.ContainingType.BaseTypeNoUseSiteDiagnostics?.IsErrorType() <> False`?

Reply:
This means we want to assert if there is no base type, hence the `False` as the second argument, which replaces the value of the first argument if it is null, otherwise the first argument is unwrapped.